src/containers/TargetScreen/index.js

```diff
@@ -0,0 +1,142 @@
+      loadTargets: false,
+      newTarget: null
+     }
+    this.handlePressMap = this.handlePressMap.bind(this);
+  }
+
+  handleCreatNewTargetClick = () => {
+    this.setState({
+      toggleNewTarget: true,
+    });
+  }
+
+  componentDidMount(){
+    this.props.getTargets().then(() => {
+      this.setState({
+        loadTargets: true,
+      });
+    }, error => {
+      console.log(error);
+    });
+  }
+
+  handlePressMap(e){
+    const { coordinate } = e.nativeEvent;
+    this.setState({
+      newTarget: { coordinate },
+      toggleNewTarget: false
+    })
+  }
+
+  renderNewTarget(){
+    const { newTarget: { coordinate } } = this.state
+    return (
+      <View>
+        <Marker 
+          coordinate={coordinate}
+          draggable
+          image={require('../../assets/pointer.png')}
+        >
+        </Marker>
+        <Circle
+          center={coordinate}
+          radius={50}
+          strokeColor={transparentYellow}
+          fillColor={white}
+        />
+      </View>
+    )
+  }
+
+  renderMarkers(){
+    const { targets } = this.props;
+    return targets.map(({ id, topicId , radius ,latitude, longitude }) => {
+      const lat = parseFloat(latitude);
+      const lng = parseFloat(longitude);
+      return (
+        <View key={id}>
+          <Marker 
+            coordinate={{latitude: lat, longitude: lng}}
+            image={getTopicIcon(topicId)}
+          >
+          </Marker>
+          <Circle
+            center={{latitude: lat, longitude: lng}}
+            radius={radius}
+            strokeColor={transparentYellow}
+            fillColor={transparentYellow}
+          />
+        </View>
+      );
+    });
+  }
+
+  render(){
+    const { createNewTarget } = this.props;
+    const { toggleNewTarget, loadTargets, newTarget  } = this.state;
+    const componenToRender = toggleNewTarget ? 
+                             <TargetForm onSubmit={(target) => createNewTarget(target.toJS(), newTarget)}/> :
+                             (<TouchableOpacity onPress={this.handleCreatNewTargetClick}>
+                                <Image source={require('../../assets/group.png')} style={styles.targetIcon}/>
+                                <Text style={styles.createTargetText}>CREATE NEW TARGET</Text>
+                             </TouchableOpacity>);
+    if (!loadTargets)
+      return <Loading />;
+    else
+      return (
+        <View style={styles.targetContainer}>
+          <View style={styles.container}>
+            <View style={[styles.mapContainer, {height: toggleNewTarget ?  '40%' : '80%'}]}>
+              <MapView
+                provider={PROVIDER_GOOGLE}
+                showsUserLocation={ true }
+                style={styles.map}
+                region={{
+                  latitude: 37.78825,
+                  longitude: -122.4324,
+                  latitudeDelta: 0.015,
+                  longitudeDelta: 0.0121
+                }}
+                onPress={this.handlePressMap}
```

Comment:
I would recommend using 
```suggestion
                onPress={() => this.handlePressMap() }
```
And removed the bind on constructor, what do you think @juanlacruz @aguscha333 ?

Reply:
I have been investigating your suggestion, the syntax of the arrow function creates a new anonymous function every time it is called, so the component will be render another time unnecessarily. Correct me if I'm wrong